storage/range_gc_queue.go

```diff
@@ -60,7 +60,7 @@ func (q *rangeGCQueue) needsLeaderLease() bool {
 // and if so at what priority. Currently all inactive ranges are
 // considered for possible GC at equal priority.
 func (q *rangeGCQueue) shouldQueue(now proto.Timestamp, rng *Range) (bool, float64) {
-	if _, expired := rng.HasLeaderLease(now); !expired {
+	if held, expired := rng.HasLeaderLease(now); held && !expired {
```

Comment:
Thanks for taking a look, @bdarnell ! You're right. I was totally misunderstanding. So, IIRC, `shouldQueue` eventually returns true since the leaser will not be updated after the node is removed from a replica list.

The test was failing just because the manual clock was not advancing.. The test passed after incrementing the manual clock.

I wonder the test itself might be still useful, but it might be just a subset of what `TestReplicateAddAndRemove` is doing...